internal/chart/v3/lint/rules/template.go

```diff
@@ -220,6 +220,13 @@ func validateTemplatesDir(templatesPath string) error {
 }
 
 func validateAllowedExtension(fileName string) error {
```

Comment:
I don't think the name `validateAllowedExtension` makes a lot of sense if it is used this way. It probably makes more sense to add a method named something like `isHelperFile` and return bool and call that instead.